scipy/stats/_rvs_sampling.py

```diff
@@ -168,3 +172,393 @@ def rvs_ratio_uniforms(pdf, umax, vmin, vmax, size=1, c=0, random_state=None):
         i += 1
 
     return np.reshape(x, size1d)
+
+
+class FastNumericalInverse():
+    r"""
+    Object representing a Fast Numerical Inverse of a probability distribution.
+
+    The initializer of `FastNumericalInverse` accepts `dist`, a frozen instance
+    of a continuous distribution, and provides an object with methods
+    that approximate `dist.ppf` and `dist.rvs`. For most distributions,
+    these methods are faster than those of `dist` itself.
+
+    Parameters
+    ----------
+    dist : scipy.stats.rv_frozen
+        Frozen distribution for which a fast numerical inverse is desired.
+    tol : float, optional
+        u-error tolerance (see Notes). The default is 1e-12.
+    max_intervals : int, optional
+        Maximum number of intervals in the cubic Hermite spline used to
+        approximate the percent point function. The default is 100000.
+
+    Attributes
+    ----------
+    intervals : int
+        The number of intervals of the interpolant.
+    midpoint_error : float
```

Comment:
Sure, that is a reasonable extension to the algorithm, as the midpoint error may not be the maximum error (er, I forget what the paper has to say about this?). It would probably not be something to do by default, as the user may not care, and the user should either be allowed to specify the number of points, the points to use, or a function that generates points to use. Maybe after the object has been created, it could have a method to refine its mesh rather than complicating the interface of the initializer. But sounds like something for a follow-up.

Reply:
Yes, the algorithm should remain as it is, it would just be a separate method that the user can run if he/she wishes to check the u-error. Comment in the paper:

> But for very "bad" densities the optimization program could be fooled as well. So it is important to make an empirical check if the approximation works with the desired precision by computing the u-error for a large sample of U(0, 1) variates

so something like 
```
def estimate_uerror(self, N=10000):
    u = N uniform points on [0, 1]
    return np.max(np.abs(u - cdf(self.H(u))))
```